app/services/ab_test.rb

```diff
@@ -0,0 +1,13 @@
+class AbTest
+  def initialize(key, percent_on)
+    @key = key
+    @percent_on = percent_on.to_s.to_i
+  end
+
+  def enabled?(session, reset)
```

Comment:
Is it possible to separate checking whether the feature is enabled and enabling the feature? Changing the state as a side effect of seeing if its enabled it seems unexpected to me (unless I'm measuring its quantum state...) Also, this looks like the only case where `FeatureManagement` is looking to any logic besides env variables to see if a feature is enabled and that may be confusing.

It seems like there are 2 levels here
1) whether the recaptcha feature is enabled for the environment (controlled by `FeatureManagement`)
2) whether the session requires the recaptcha (controlled by AbTest`)

What do you think about asking `AbTest` if its enabled and `AbTest` asking `FeatureManagement` instead of the other way around? Then maybe enabling it explicitly as a second step?